Docker image 2021.10.0.dev20210911 destroys MariaDB database

### The problem

After pulling the docker image for
homeassistant/raspberrypi4-homeassistant
version 2021.10.0.dev20210911
org.opencontainers.image.created | 2021-09-11 05:50:34+00:00 

A notification is showing that a database update is running.
After some time, the notification is gone, but the database isn't any more connected.
The database container I'm running is showing that the database is corrupt and is starting over and over again.

After deleting and restoring a backup from the night the database is up and running until I start the container that is making the update, with the yesterday container there is no error.

### What is version of Home Assistant Core has the issue?

2021.10.0.dev20210911

### What was the last working version of Home Assistant Core?

2021.10.0.dev20210908

### What type of installation are you running?

Home Assistant Container

### Integration causing the issue

_No response_

### Link to integration documentation on our website

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

Okay, spoken too soon.

The notification was gone for about 3 minutes, than it crashes

HA recorder error logs:
```
2021-09-13 21:44:44 ERROR (Recorder) [homeassistant.components.recorder] Database connection invalidated: Error executing query: (MySQLdb._exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query')
[SQL: INSERT INTO states (domain, entity_id, state, attributes, event_id, last_changed, last_updated, created, old_state_id) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)]
[parameters: ('sensor', 'sensor.power_apparent_phase_2_fronius_meter_1_http_192_168_178_38', '402.09', '{"state_class":"measurement","unit_of_measurement":"VA","friendly_name":"Power apparent phase 2 Fronius Meter 1 http://192.168.178.38","device_class":"power"}', 148868027, datetime.datetime(2021, 9, 13, 19, 36, 52, 735075, tzinfo=datetime.timezone.utc), datetime.datetime(2021, 9, 13, 19, 36, 52, 735075, tzinfo=datetime.timezone.utc), datetime.datetime(2021, 9, 13, 19, 36, 52, 735075, tzinfo=datetime.timezone.utc), 147715811)]
(Background on this error at: https://sqlalche.me/e/14/e3q8). (retrying in 3 seconds)
2021-09-13 21:44:47 ERROR (Recorder) [homeassistant.components.recorder] SQLAlchemyError error processing event <Event time_changed[L]: now=2021-09-13T21:36:53.000866+02:00>: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (MySQLdb._exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query')
[SQL: INSERT INTO states (domain, entity_id, state, attributes, event_id, last_changed, last_updated, created, old_state_id) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)]
[parameters: ('sensor', 'sensor.power_apparent_phase_2_fronius_meter_1_http_192_168_178_38', '402.09', '{"state_class":"measurement","unit_of_measurement":"VA","friendly_name":"Power apparent phase 2 Fronius Meter 1 http://192.168.178.38","device_class":"power"}', 148868027, datetime.datetime(2021, 9, 13, 19, 36, 52, 735075, tzinfo=datetime.timezone.utc), datetime.datetime(2021, 9, 13, 19, 36, 52, 735075, tzinfo=datetime.timezone.utc), datetime.datetime(2021, 9, 13, 19, 36, 52, 735075, tzinfo=datetime.timezone.utc), 147715811)]
(Background on this error at: https://sqlalche.me/e/14/e3q8) (Background on this error at: https://sqlalche.me/e/14/7s2a)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 657, in _process_one_event_or_recover
    self._process_one_event(event)
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 777, in _process_one_event
    self._commit_event_session_or_retry()
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 840, in _commit_event_session_or_retry
    self._commit_event_session()
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 866, in _commit_event_session
    self.event_session.commit()
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 1428, in commit
    self._transaction.commit(_to_root=self.future)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 827, in commit
    self._assert_active(prepared_ok=True)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 601, in _assert_active
    raise sa_exc.PendingRollbackError(
sqlalchemy.exc.PendingRollbackError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (MySQLdb._exceptions.OperationalError) (2013, 'Lost connection to MySQL server during query')
[SQL: INSERT INTO states (domain, entity_id, state, attributes, event_id, last_changed, last_updated, created, old_state_id) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)]
[parameters: ('sensor', 'sensor.power_apparent_phase_2_fronius_meter_1_http_192_168_178_38', '402.09', '{"state_class":"measurement","unit_of_measurement":"VA","friendly_name":"Power apparent phase 2 Fronius Meter 1 http://192.168.178.38","device_class":"power"}', 148868027, datetime.datetime(2021, 9, 13, 19, 36, 52, 735075, tzinfo=datetime.timezone.utc), datetime.datetime(2021, 9, 13, 19, 36, 52, 735075, tzinfo=datetime.timezone.utc), datetime.datetime(2021, 9, 13, 19, 36, 52, 735075, tzinfo=datetime.timezone.utc), 147715811)]
(Background on this error at: https://sqlalche.me/e/14/e3q8) (Background on this error at: https://sqlalche.me/e/14/7s2a)
2021-09-13 21:44:49 ERROR (Recorder) [homeassistant.components.recorder] Error in database connectivity during commit: Error executing query: (MySQLdb._exceptions.OperationalError) (2002, "Can't connect to local MySQL server through socket '/opt/mysqld/mysqld.sock' (111)")
(Background on this error at: https://sqlalche.me/e/14/e3q8). (retrying in 3 seconds)
2021-09-13 21:44:52 ERROR (Recorder) [homeassistant.components.recorder] SQLAlchemyError error processing event <Event time_changed[L]: now=2021-09-13T21:36:56.001498+02:00>: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (MySQLdb._exceptions.OperationalError) (2002, "Can't connect to local MySQL server through socket '/opt/mysqld/mysqld.sock' (111)")
(Background on this error at: https://sqlalche.me/e/14/e3q8) (Background on this error at: https://sqlalche.me/e/14/7s2a)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 657, in _process_one_event_or_recover
    self._process_one_event(event)
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 777, in _process_one_event
    self._commit_event_session_or_retry()
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 840, in _commit_event_session_or_retry
    self._commit_event_session()
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 866, in _commit_event_session
    self.event_session.commit()
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 1428, in commit
    self._transaction.commit(_to_root=self.future)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 827, in commit
    self._assert_active(prepared_ok=True)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 601, in _assert_active
    raise sa_exc.PendingRollbackError(
sqlalchemy.exc.PendingRollbackError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (MySQLdb._exceptions.OperationalError) (2002, "Can't connect to local MySQL server through socket '/opt/mysqld/mysqld.sock' (111)")
(Background on this error at: https://sqlalche.me/e/14/e3q8) (Background on this error at: https://sqlalche.me/e/14/7s2a)
2021-09-13 21:44:52 ERROR (Recorder) [homeassistant.components.recorder] Error in database connectivity during commit: Error executing query: (MySQLdb._exceptions.OperationalError) (2002, "Can't connect to local MySQL server through socket '/opt/mysqld/mysqld.sock' (111)")
(Background on this error at: https://sqlalche.me/e/14/e3q8). (retrying in 3 seconds)
2021-09-13 21:44:55 ERROR (Recorder) [homeassistant.components.recorder] SQLAlchemyError error processing event <Event time_changed[L]: now=2021-09-13T21:36:57.001039+02:00>: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (MySQLdb._exceptions.OperationalError) (2002, "Can't connect to local MySQL server through socket '/opt/mysqld/mysqld.sock' (111)")
(Background on this error at: https://sqlalche.me/e/14/e3q8) (Background on this error at: https://sqlalche.me/e/14/7s2a)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 657, in _process_one_event_or_recover
    self._process_one_event(event)
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 777, in _process_one_event
    self._commit_event_session_or_retry()
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 840, in _commit_event_session_or_retry
    self._commit_event_session()
  File "/usr/src/homeassistant/homeassistant/components/recorder/__init__.py", line 866, in _commit_event_session
    self.event_session.commit()
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 1428, in commit
    self._transaction.commit(_to_root=self.future)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 827, in commit
    self._assert_active(prepared_ok=True)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 601, in _assert_active
    raise sa_exc.PendingRollbackError(
sqlalchemy.exc.PendingRollbackError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (MySQLdb._exceptions.OperationalError) (2002, "Can't connect to local MySQL server through socket '/opt/mysqld/mysqld.sock' (111)")

```

And the DB logs:
```
[i] mysqld already present, skipping creation
[i] MySQL directory already present, skipping creation
2021-09-13 21:12:42 0 [Note] /usr/bin/mysqld (mysqld 10.5.11-MariaDB) starting as process 1 ...
2021-09-13 21:12:42 0 [Note] InnoDB: Uses event mutexes
2021-09-13 21:12:42 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-09-13 21:12:42 0 [Note] InnoDB: Number of pools: 1
2021-09-13 21:12:42 0 [Note] InnoDB: Using generic crc32 instructions
2021-09-13 21:12:42 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
2021-09-13 21:12:42 0 [Note] InnoDB: Using Linux native AIO
2021-09-13 21:12:42 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
2021-09-13 21:12:43 0 [Note] InnoDB: Completed initialization of buffer pool
2021-09-13 21:12:43 0 [Note] InnoDB: 128 rollback segments are active.
2021-09-13 21:12:43 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2021-09-13 21:12:43 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2021-09-13 21:12:43 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2021-09-13 21:12:43 0 [Note] InnoDB: 10.5.11 started; log sequence number 9848336337; transaction id 208970
2021-09-13 21:12:43 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-09-13 21:12:43 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
2021-09-13 21:12:43 0 [Note] Server socket created on IP: '::'.
2021-09-13 21:12:43 0 [Warning] 'user' entry '@ab9aafc7d18b' ignored in --skip-name-resolve mode.
2021-09-13 21:12:43 0 [Note] Reading of all Master_info entries succeeded
2021-09-13 21:12:43 0 [Note] Added new Master_info '' to hash table
2021-09-13 21:12:43 0 [Note] /usr/bin/mysqld: ready for connections.
Version: '10.5.11-MariaDB'  socket: '/run/mysqld/mysqld.sock'  port: 3306  MariaDB Server
2021-09-13 21:12:44 0 [Note] InnoDB: Buffer pool(s) load completed at 210913 21:12:44
2021-09-13 21:44:43 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2130], should be [page id: space=11, page number=264274]
2021-09-13 21:44:43 0x9a183d30  InnoDB: Assertion failure in file /home/buildozer/aports/main/mariadb/src/mariadb-10.5.11/storage/innobase/btr/btr0cur.cc line 335
InnoDB: Failing assertion: btr_page_get_prev(get_block->frame) == block->page.id().page_no()
InnoDB: We intentionally generate a memory trap.
InnoDB: Submit a detailed bug report to https://jira.mariadb.org/
InnoDB: If you get repeated assertion failures or crashes, even
InnoDB: immediately after the mysqld startup, there may be
InnoDB: corruption in the InnoDB tablespace. Please refer to
InnoDB: https://mariadb.com/kb/en/library/innodb-recovery-modes/
InnoDB: about forcing recovery.
[i] mysqld already present, skipping creation
[i] MySQL directory already present, skipping creation
2021-09-13 21:44:55 0 [Note] /usr/bin/mysqld (mysqld 10.5.11-MariaDB) starting as process 1 ...
2021-09-13 21:44:55 0 [Note] InnoDB: Uses event mutexes
2021-09-13 21:44:55 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-09-13 21:44:55 0 [Note] InnoDB: Number of pools: 1
2021-09-13 21:44:55 0 [Note] InnoDB: Using generic crc32 instructions
2021-09-13 21:44:55 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
2021-09-13 21:44:55 0 [Note] InnoDB: Using Linux native AIO
2021-09-13 21:44:55 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
2021-09-13 21:44:55 0 [Note] InnoDB: Completed initialization of buffer pool
2021-09-13 21:44:55 0 [Note] InnoDB: Starting crash recovery from checkpoint LSN=11342390579,11353482681
2021-09-13 21:44:59 0 [Note] InnoDB: Starting final batch to recover 4207 pages from redo log.
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=0], should be [page id: space=11, page number=262144]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=1], should be [page id: space=11, page number=262145]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=1387], should be [page id: space=11, page number=263531]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Not applying INSERT_HEAP_DYNAMIC due to corruption on [page id: space=11, page number=263531]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Set innodb_force_recovery=1 to ignore corruption.
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=1761], should be [page id: space=11, page number=263905]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=1787], should be [page id: space=11, page number=263931]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=1958], should be [page id: space=11, page number=264102]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=1979], should be [page id: space=11, page number=264123]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=1980], should be [page id: space=11, page number=264124]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=1988], should be [page id: space=11, page number=264132]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=1993], should be [page id: space=11, page number=264137]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2001], should be [page id: space=11, page number=264145]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2010], should be [page id: space=11, page number=264154]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2026], should be [page id: space=11, page number=264170]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2027], should be [page id: space=11, page number=264171]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2032], should be [page id: space=11, page number=264176]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2054], should be [page id: space=11, page number=264198]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2072], should be [page id: space=11, page number=264216]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2073], should be [page id: space=11, page number=264217]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2077], should be [page id: space=11, page number=264221]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2100], should be [page id: space=11, page number=264244]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2097], should be [page id: space=11, page number=264241]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2123], should be [page id: space=11, page number=264267]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2134], should be [page id: space=11, page number=264278]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2140], should be [page id: space=11, page number=264284]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2142], should be [page id: space=11, page number=264286]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2125], should be [page id: space=11, page number=264269]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2129], should be [page id: space=11, page number=264273]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2144], should be [page id: space=11, page number=264288]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2143], should be [page id: space=11, page number=264287]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2145], should be [page id: space=11, page number=264289]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2160], should be [page id: space=11, page number=264304]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2161], should be [page id: space=11, page number=264305]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2131], should be [page id: space=11, page number=264275]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2162], should be [page id: space=11, page number=264306]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2163], should be [page id: space=11, page number=264307]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2164], should be [page id: space=11, page number=264308]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2165], should be [page id: space=11, page number=264309]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2166], should be [page id: space=11, page number=264310]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2167], should be [page id: space=11, page number=264311]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2169], should be [page id: space=11, page number=264313]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2173], should be [page id: space=11, page number=264317]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2177], should be [page id: space=11, page number=264321]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2193], should be [page id: space=11, page number=264337]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2194], should be [page id: space=11, page number=264338]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2196], should be [page id: space=11, page number=264340]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2197], should be [page id: space=11, page number=264341]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2199], should be [page id: space=11, page number=264343]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2200], should be [page id: space=11, page number=264344]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2202], should be [page id: space=11, page number=264346]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2203], should be [page id: space=11, page number=264347]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2204], should be [page id: space=11, page number=264348]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2205], should be [page id: space=11, page number=264349]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2206], should be [page id: space=11, page number=264350]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2208], should be [page id: space=11, page number=264352]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2212], should be [page id: space=11, page number=264356]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2215], should be [page id: space=11, page number=264359]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2216], should be [page id: space=11, page number=264360]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2217], should be [page id: space=11, page number=264361]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2218], should be [page id: space=11, page number=264362]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2219], should be [page id: space=11, page number=264363]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2220], should be [page id: space=11, page number=264364]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2221], should be [page id: space=11, page number=264365]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2226], should be [page id: space=11, page number=264370]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2229], should be [page id: space=11, page number=264373]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2231], should be [page id: space=11, page number=264375]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2233], should be [page id: space=11, page number=264377]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2234], should be [page id: space=11, page number=264378]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2284], should be [page id: space=11, page number=264428]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2285], should be [page id: space=11, page number=264429]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2300], should be [page id: space=11, page number=264444]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2305], should be [page id: space=11, page number=264449]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2315], should be [page id: space=11, page number=264459]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2317], should be [page id: space=11, page number=264461]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2336], should be [page id: space=11, page number=264480]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2338], should be [page id: space=11, page number=264482]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2339], should be [page id: space=11, page number=264483]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2340], should be [page id: space=11, page number=264484]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2341], should be [page id: space=11, page number=264485]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2342], should be [page id: space=11, page number=264486]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2351], should be [page id: space=11, page number=264495]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2352], should be [page id: space=11, page number=264496]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2354], should be [page id: space=11, page number=264498]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2362], should be [page id: space=11, page number=264506]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2375], should be [page id: space=11, page number=264519]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=2376], should be [page id: space=11, page number=264520]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=4526], should be [page id: space=11, page number=266670]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Not applying INSERT_HEAP_DYNAMIC due to corruption on [page id: space=11, page number=266670]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Set innodb_force_recovery=1 to ignore corruption.
2021-09-13 21:45:04 0 [ERROR] InnoDB: Not applying INSERT_HEAP_DYNAMIC due to corruption on [page id: space=11, page number=266670]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Set innodb_force_recovery=1 to ignore corruption.
2021-09-13 21:45:04 0 [ERROR] InnoDB: Space id and page no stored in the page, read in are [page id: space=11, page number=4528], should be [page id: space=11, page number=266672]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Not applying INSERT_HEAP_DYNAMIC due to corruption on [page id: space=11, page number=266672]
2021-09-13 21:45:04 0 [ERROR] InnoDB: Set innodb_force_recovery=1 to ignore corruption.
2021-09-13 21:45:04 0 [ERROR] InnoDB: Plugin initialization aborted with error Data structure corruption
2021-09-13 21:45:04 0 [Note] InnoDB: Starting shutdown...
2021-09-13 21:45:07 0 [ERROR] Plugin 'InnoDB' init function returned error.
2021-09-13 21:45:07 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2021-09-13 21:45:07 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-09-13 21:45:07 0 [ERROR] Unknown/unsupported storage engine: InnoDB
2021-09-13 21:45:07 0 [ERROR] Aborting
```

--- a/homeassistant/components/recorder/migration.py
+++ b/homeassistant/components/recorder/migration.py
@@ -522,9 +522,11 @@ def _apply_update(instance, session, new_version, old_version):  # noqa: C901
                 )
                 with contextlib.suppress(SQLAlchemyError):
                     connection.execute(
+                        # Using LOCK=EXCLUSIVE to prevent the database from corrupting
+                        # https://github.com/home-assistant/core/issues/56104
                         text(
                             f"ALTER TABLE {table} CONVERT TO "
-                            "CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci"
+                            "CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci LOCK=EXCLUSIVE"
                         )
                     )
     elif new_version == 22:

